helix-core/src/main/java/org/apache/helix/controller/rebalancer/strategy/AutoRebalanceStrategy.java

```diff
@@ -336,13 +347,21 @@ private void prepareResult(ZNRecord znRecord) {
         }
       }
     }
-    normalizePreferenceLists(znRecord.getListFields(), newPreferences);
+    normalizePreferenceLists(znRecord.getListFields(), newPreferences, clusterData);
+
+    String stateModelDef = clusterData.getIdealState(_resourceName).getStateModelDefRef();
+    StateModelDefinition stateModel = clusterData.getStateModelDef(stateModelDef);
 
     // generate preference maps based on the preference lists
     for (String partition : _partitions) {
       List<String> preferenceList = znRecord.getListField(partition);
       int i = 0;
       for (String participant : preferenceList) {
+        // if the participant is not alive, assign it to the initial state
```

Comment:
If the node is not alive, why we have in the map? I assume we fix disabled ones, right?

My assumption is if the node is not alive, we should not show something in map field as this is strategy, this is not delay auto.

Only delay auto consider offline instances with offline timeout and put it as offline when node is not alive.